### PR TITLE
Improve MOC generation

### LINKED ISSUE #322: Improve MOC generation
Allow users to generate MOCs without writing them to file. Also allow `gen_mocs_epoch` to use any STMOC file.

### PR DESCRIPTION
**Pull request by ddobie**
Fix #322.

### CODE REVIEW

File: tests/test_tools.py [hunk 1]
```diff
@@ -336,6 +342,7 @@ def test_gen_mocs_image(
             FITS file.
         dummy_load_fields_file: The dummy fields file.
         dummy_moc: The dummy MOC object representing an open MOC.
+        write: Whether to test the write to file or not
```
Comment (ajstewart) [inline tests/test_tools.py:345]:
```suggestion
        write: Whether to test the write to file or not.
```
Vital 😛

File: vasttools/tools.py [hunk 2]
```diff
@@ -471,8 +475,17 @@ def gen_mocs_epoch(epoch: str,
...
-    full_STMOC = vtm.load_pilot_stmoc()
+    if base_stmoc is None:
+        vtm = VASTMOCS()
+        full_STMOC = vtm.load_pilot_stmoc()
+    else:
+        if isinstance(base_stmoc, str):
+            base_stmoc = Path(base_stmoc)
+
+        if not base_stmoc.exists():
+            raise Exception("{} does not exist".format(base_stmoc))
```
Comment (ajstewart) [inline vasttools/tools.py:486]: You could check in the tests that this error is called correctly? If not too much work, otherwise it's ok.

**Comment by ajstewart** [comment]
Do any docs need updating to highlight the change/feature?

**Comment by ddobie** [comment]
Yeah good point. The previous docs weren't great and were ambiguous enough that they didn't need to be changed, but I've added a little more detail.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Added tools for new epoch addition: add datetimes to fits images create fields csv files, create and update MOCs [#298](https://github.com/askap-vast/vast-tools/pull/298)
 
 #### Changed
+- Allow users to specify whether to write generated MOCs, and to use arbitrary STMOC [#322](https://github.com/askap-vast/vast-tools/pull/323)
```
Comment (ajstewart) [inline CHANGELOG.md:15]:
```suggestion
- Allow users to specify whether to write generated MOCs, and to use arbitrary STMOC [#323](https://github.com/askap-vast/vast-tools/pull/323)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -20,6 +21,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 #### Removed
 
 #### List of PRs
+- [#322](https://github.com/askap-vast/vast-tools/pull/323): feat, tests: New options for MOC generation
```
Comment (ajstewart) [inline CHANGELOG.md:24]:
```suggestion
- [#323](https://github.com/askap-vast/vast-tools/pull/323): feat, tests: New options for MOC generation
```

**Review by ajstewart — DISMISSED** [review]
Looks good, there are a couple minor corrections to the changelog numbering.

**Comment by ddobie** [comment]
Weird. Not sure how I missed that!

**Pull request merged**